### DISCUSSION TITLE
【LMQ】Does LMQ have a quantity limit?

### DISCUSSION THREAD

**Discussion by humkum** [apache/rocketmq#6899] [General]
In the early stages of LMQ design，there used to be the limit of LMQ quantity, which was 20,000 every broker defaulted. But the logic of LMQ quantity limit had been removed later. 
Why did we remove this restriction? hould we consider the LMQ’s quantity in using? And what is the effect of too many LMQs？

![image](https://github.com/apache/rocketmq/assets/50660789/9fbaa44b-b060-4e3c-a90f-688b651ee35f)

![image](https://github.com/apache/rocketmq/assets/50660789/e0a7dae4-0dfc-48fc-8c2b-72c63a1b91ce)
![image](https://github.com/apache/rocketmq/assets/50660789/b682a677-006e-426c-8172-ab6e5f8984c8)
![image](https://github.com/apache/rocketmq/assets/50660789/0410e671-2096-4ce2-acd5-097cc105cc59)

**Comment by humkum** [comment]
@tianliuliu I'm looking forward to your reply

**Reply by tianliuliu** [reply]
> @tianliuliu I'm looking forward to your reply
@humkum 
1. The early design did set a limit on the number of LMQs to 20,000. As for the subsequent deletion of the limit, I am not sure. This can be traced to the code git push record and asked about the reason for the deletion of the limit.
2. The introduction of LMQ will lead to the introduction of a large number of small index files into the RocketMQ system. Considering the limitation of the operating system file system inode, and the maintenance and dataStore of metadata is a very big challenge, so this limitation is added.
3. In the usage scenario, it is not recommended to open too many LMQs. You can adjust the grayscale step by step, and check the system load to finally determine the number to use. The current 20,000 is just an experience value.

**Reply by humkum** [reply]
thx
